icis-air-inspections.ipynb

```diff
@@ -619,53 +413,13 @@
-      "text/plain": [
-       "<Figure size 1080x720 with 1 Axes>"
-      ]
-     },
-     "metadata": {
-      "needs_background": "light"
-     },
-     "output_type": "display_data"
-    }
-   ],
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
    "source": [
     "chart_title = \"Total CAA inspections\"\n",
     "\n",
-    "show_plot( state_icis_fac_data, 'ACTUAL_END_DATE', 'ACTUAL_END_YEAR', my_state, \\\n",
+    "show_plot( state_icis_fac_data, 'ACTUAL_END_DATE', my_state, \\\n",
```

Comment:
Got an error:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_slice_bound(self, label, side, kind)
   5166             try:
-> 5167                 return self._searchsorted_monotonic(label, side)
   5168             except ValueError:

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/base.py in _searchsorted_monotonic(self, label, side)
   5127 
-> 5128         raise ValueError("index must be monotonic increasing or decreasing")
   5129 

ValueError: index must be monotonic increasing or decreasing

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)
<ipython-input-16-64490a60e7f8> in <module>
      2 
      3 show_plot( state_icis_fac_data, 'ACTUAL_END_DATE', my_state, \
----> 4           '%m/%d/%Y', chart_title )

<ipython-input-15-2972f09f77f9> in show_plot(df, date_field, place, date_format, chart_title)
      5 def show_plot( df, date_field, place, date_format, chart_title ):
      6 
----> 7     df[date_field] = pd.to_datetime(df[date_field], format=date_format) #format
      8 
      9     cy = df.groupby([date_field])['ACTUAL_END_DATE'].count() # Use a dummy variable to count inspections.

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    206                 else:
    207                     kwargs[new_arg_name] = new_arg_value
--> 208             return func(*args, **kwargs)
    209 
    210         return wrapper

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in to_datetime(arg, errors, dayfirst, yearfirst, utc, box, format, exact, unit, infer_datetime_format, origin, cache)
    772                 result = result.tz_localize(tz)
    773     elif isinstance(arg, ABCSeries):
--> 774         cache_array = _maybe_cache(arg, format, cache, convert_listlike)
    775         if not cache_array.empty:
    776             result = arg.map(cache_array)

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in _maybe_cache(arg, format, cache, convert_listlike)
    149     if cache:
    150         # Perform a quicker unique check
--> 151         if not should_cache(arg):
    152             return cache_array
    153 

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/tools/datetimes.py in should_cache(arg, unique_share, check_count)
    119     assert 0 < unique_share < 1, "unique_share must be in next bounds: (0; 1)"
    120 
--> 121     unique_elements = unique(arg[:check_count])
    122     if len(unique_elements) > check_count * unique_share:
    123         do_caching = False

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/series.py in __getitem__(self, key)
   1111             key = check_bool_indexer(self.index, key)
   1112 
-> 1113         return self._get_with(key)
   1114 
   1115     def _get_with(self, key):

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/series.py in _get_with(self, key)
   1116         # other: fancy integer or otherwise
   1117         if isinstance(key, slice):
-> 1118             indexer = self.index._convert_slice_indexer(key, kind="getitem")
   1119             return self._get_values(indexer)
   1120         elif isinstance(key, ABCDataFrame):

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/numeric.py in _convert_slice_indexer(self, key, kind)
    395 
    396         # translate to locations
--> 397         return self.slice_indexer(key.start, key.stop, key.step, kind=kind)
    398 
    399     def _format_native_types(

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/base.py in slice_indexer(self, start, end, step, kind)
   5032         slice(1, 3)
   5033         """
-> 5034         start_slice, end_slice = self.slice_locs(start, end, step=step, kind=kind)
   5035 
   5036         # return a slice

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/base.py in slice_locs(self, start, end, step, kind)
   5252         end_slice = None
   5253         if end is not None:
-> 5254             end_slice = self.get_slice_bound(end, "right", kind)
   5255         if end_slice is None:
   5256             end_slice = len(self)

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_slice_bound(self, label, side, kind)
   5168             except ValueError:
   5169                 # raise the original KeyError
-> 5170                 raise err
   5171 
   5172         if isinstance(slc, np.ndarray):

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_slice_bound(self, label, side, kind)
   5162         # we need to look up the label
   5163         try:
-> 5164             slc = self.get_loc(label)
   5165         except KeyError as err:
   5166             try:

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/numeric.py in get_loc(self, key, method, tolerance)
    477         except (TypeError, NotImplementedError):
    478             pass
--> 479         return super().get_loc(key, method=method, tolerance=tolerance)
    480 
    481     @cache_readonly

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   2897                 return self._engine.get_loc(key)
   2898             except KeyError:
-> 2899                 return self._engine.get_loc(self._maybe_cast_indexer(key))
   2900         indexer = self.get_indexer([key], method=method, tolerance=tolerance)
   2901         if indexer.ndim > 1 or indexer.size > 1:

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine._get_loc_duplicates()

pandas/_libs/index_class_helper.pxi in pandas._libs.index.Float64Engine._maybe_get_bool_indexer()

KeyError: 26
```

for NY/1

Reply:
Yikes, taking a look now!

Reply:
Hmm I was unable to reproduce this error locally or on this Colab version of the branch: https://colab.research.google.com/github/edgi-govdata-archiving/ECHO-Clean-Air-Act/blob/chart-change/icis-air-inspections.ipynb I'm not sure why 26 is a key in this case. Any thoughts??!?

Reply:
Hm, no longer seems to be a problem. Wonder what that was!